library/src/main/java/com/prolificinteractive/materialcalendarview/MonthPagerAdapter.java

```diff
@@ -0,0 +1,303 @@
+package com.prolificinteractive.materialcalendarview;
+
+import android.support.annotation.Nullable;
+import android.support.v4.view.PagerAdapter;
+import android.view.View;
+import android.view.ViewGroup;
+
+import com.prolificinteractive.materialcalendarview.format.DayFormatter;
+import com.prolificinteractive.materialcalendarview.format.WeekDayFormatter;
+
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * Pager adapter backing the calendar view
+ */
+class MonthPagerAdapter extends PagerAdapter {
+
+    private final LinkedList<MonthView> currentViews;
+    private final ArrayList<CalendarDay> months;
```

Comment:
It's a list of the first days of months. I'm working on a more efficient way of paging months in the [aek/month-math](https://github.com/prolificinteractive/material-calendarview/compare/aek/custom-views...aek/month-math) branch, so this should become moot.